src/status_im/contexts/wallet/wallet_connect/effects.cljs

```diff
@@ -41,10 +46,47 @@
  :effects.wallet-connect/approve-session
  (fn [{:keys [web3-wallet proposal supported-namespaces on-success on-fail]}]
    (let [{:keys [params id]} proposal
-         approved-namespaces (wallet-connect/build-approved-namespaces params
-                                                                       supported-namespaces)]
+         approved-namespaces (wallet-connect/build-approved-namespaces
+                              params
+                              supported-namespaces)]
      (-> (.approveSession web3-wallet
                           (clj->js {:id         id
                                     :namespaces approved-namespaces}))
          (promesa/then on-success)
          (promesa/catch on-fail)))))
+
+(rf/reg-fx
+ :effects.wallet-connect/sign-message
+ (fn [{:keys [password address data on-success on-error]}]
+   (-> {:data     data
+        :account  address
+        :password (security/safe-unmask-data password)}
+       bean/->js
+       transforms/clj->json
+       native-module/sign-message
+       (promesa/then wallet-connect-core/extract-native-call-signature)
+       (promesa/then on-success)
+       (promesa/catch on-error))))
+
+(rf/reg-fx
+ :effects.wallet-connect/sign-typed-data
+ (fn [{:keys [password address data version on-success on-error]}]
+   (-> {:v1 native-module/sign-typed-data
+        :v4 native-module/sign-typed-data-v4}
+       (get version)
+       (apply [data address (security/safe-unmask-data password)])
```

Comment:
This could be its own separate function that's callable based on the version. It's a bit odd to read what could be a normal function call a sequence of operations with the `apply` at the end.

Suggestion (untested). I kept the vars in the same namespace for convenience for the snippet. `sign-typed-data` is the kind of function that could be instrumented by malli because if the versions passed are unsupported, everything explodes.

```clojure
(def ^:private sign-typed-data-by-version
  {:v1 native-module/sign-typed-data
   :v4 native-module/sign-typed-data-v4})

(defn sign-typed-data
  [version data address password]
  (let [f (get sign-typed-data-by-version version)]
    (f data address (security/safe-unmask-data password))))

(rf/reg-fx :effects.wallet-connect/sign-typed-data
 (fn [{:keys [password address data version on-success on-error]}]
   (-> (sign-typed-data version data address password)
       (promesa/then wallet-connect-core/extract-native-call-signature)
       (promesa/then on-success)
       (promesa/catch on-error))))
```

Reply:
yes it definitely reads better. Thanks!

Reply:
done